public/assets/js/dropzoneFunctions.js

```diff
@@ -1,53 +1,110 @@
+            primaryDropzone.lastElementChild.setAttribute('class', '');
 
-function dragover_handler(event) {
-    event.preventDefault();
-}
-
-function dragend_handler(event) {
-    var dt = event.dataTransfer;
-    if (dt.items) {
-        for (var i = 0; i < dt.items.length; i++) {
-            dt.items.remove(i);
+        },
+        dragexit_handler: function () {
+            var primaryDropzone = document.getElementById('primary-dropzone');
+            primaryDropzone.setAttribute('class', 'dropzone row row--tall row--margined row--padded flex-container--column flex-container--center-center');
+            primaryDropzone.firstElementChild.setAttribute('class', '');
+            primaryDropzone.lastElementChild.setAttribute('class', 'hidden');
+        },
+        showError: function (errorMsg) {
+            const errorDisplay = document.getElementById('input-error-file-selection');
+            errorDisplay.hidden = false;
+            errorDisplay.innerText = errorMsg;
+        },
+    },
+    preview: {
+        drop_handler: function (event) {
+            event.preventDefault();
+            var dt = event.dataTransfer;
+            if (dt.items) {
+                if (dt.items[0].kind == 'file') {
+                    var droppedFile = dt.items[0].getAsFile();
+                    this.selectFile(droppedFile);
+                }
+            }
+        },
+        selectFile(selectedFile){
+            this.onmouseleave_handler();
+            try {
+                validationFunctions.validateFile(selectedFile); // validate the file's name, type, and size
+            } catch (error) {
+                this.showError(error.message);
+                return;
+            }
+            publishFileFunctions.previewAndStageFile(selectedFile);
+        },
+        onmouseenter_handler: function () {
+            // show drag-and-drop instructions
+            document.getElementById('asset-preview-dropzone-instructions').setAttribute('class', 'flex-container--column flex-container--center-center position-absolute');
+            // make image lighter
+            document.getElementById('asset-preview').style.opacity = 0.2;
+        },
+        onmouseleave_handler: function () {
+            // hide instructions
+            document.getElementById('asset-preview-dropzone-instructions').setAttribute('class', 'hidden');
+            // make image regular visibility
+            document.getElementById('asset-preview').style.opacity = 1;
+        },
+        showError: function(errorMsg){
+            this.onmouseleave_handler();
+            // show error
+            const errorElement = document.getElementById('asset-preview-dropzone-error');
+            errorElement.setAttribute('class', 'flex-container--column flex-container--center-center position-absolute');
+            errorElement.innerHTML = `<p>${errorMsg}</p>`;
+            // set timer to hide error
+            setTimeout(this.hideError, 2000);
```

Comment:
While this is a bit of a corner case, this pattern can technically introduce a confusing situation where an error flashes extremely briefly. If I manage to generate 2 errors within the timeout window, the second error will be cleared by the first callback.

Additionally, I'm not a huge fan of clearing errors on a time-basis. It's generally preferable to clear the error once the user has taken an action that can be presumed to have corrected it (e.g. typing in the relevant field, selecting a new file, whatever).

(Neither of these necessarily have to be addressed to merge this, but you should at least be aware.)